# HG changeset patch
# User Magnus Melin <mkmelin+mozilla@iki.fi>
# Date 1752878929 -7200
# Node ID a360edd30358813898aac1deaa9bbf7ce70b51ff
# Parent  04289e958b386a2d864fbd58b88b6e153e73882d
Bug 1971303 - Fix some docs and inconsistencies for showNotification(). r=vineet,BenB

The actual error was not provided for account-hub-addon-error.
Not finding an config is not an error.
Fixed docs to state what arguments are optional.
In one instance error?.message was checked but then error.message accessed. Could blow up if so.

Differential Revision: https://phabricator.services.mozilla.com/D253941

diff --git a/mail/components/accountcreation/content/widgets/account-hub-header.mjs b/mail/components/accountcreation/content/widgets/account-hub-header.mjs
--- a/mail/components/accountcreation/content/widgets/account-hub-header.mjs
+++ b/mail/components/accountcreation/content/widgets/account-hub-header.mjs
@@ -132,17 +132,16 @@ class AccountHubHeader extends HTMLEleme
         descriptionElement.querySelector(".localized-description"),
         fluentDescriptionId || error.cause.fluentDescriptionId
       );
 
-      // If we have a specific fluent ID for the description, return early
+      // If we have a specific Fluent id for the description, return early
       // so we don't have two descriptions.
       return;
     }
 
-    if (description || (type === "error" && error?.message)) {
-      let descriptionText = `${description || ""}`;
-
-      if (type === "error" && error.message) {
+    if (description || (type == "error" && error?.message)) {
+      let descriptionText = description || "";
+      if (type == "error" && error?.message) {
         descriptionText += `${descriptionText ? " - " : ""}${error.message}`;
       }
 
       descriptionElement.querySelector(".raw-description").textContent =
diff --git a/mail/components/accountcreation/views/email.mjs b/mail/components/accountcreation/views/email.mjs
--- a/mail/components/accountcreation/views/email.mjs
+++ b/mail/components/accountcreation/views/email.mjs
@@ -538,8 +538,9 @@ class AccountHubEmail extends HTMLElemen
           this.#emailFooter.toggleForwardDisabled(false);
         } catch (error) {
           this.#currentSubview.showNotification({
             fluentTitleId: "account-hub-addon-error",
+            error,
             type: "error",
           });
         }
         this.#stopLoading();
@@ -836,9 +837,9 @@ class AccountHubEmail extends HTMLElemen
             // show an error.
             this.#initUI(this.#states[this.#currentState].previousStep);
             this.#currentSubview.showNotification({
               fluentTitleId: "account-hub-find-account-settings-failed",
-              type: "error",
+              type: "warning",
             });
           }
         } catch (error) {
           this.#stopLoading();
